### DISCUSSION TITLE
Should PrefixUpTo and PrefixThrough work with @ParserBuilder as input ?

### DISCUSSION THREAD

**Discussion by mackoj** [pointfreeco/swift-parsing#164] [Q&A] [answered]
Do you think that being able to passe multiple parser to **PrefixUpTo** and **PrefixThrough** is a desirable feature ? Or will it lead to bad utilisation of the library ?

```swift
PrefixUpTo {
  ","
  Newline()
  End()
}
```

**Comment by randomeizer** [comment, marked as answer]
This was something I asked a while back, and the concern from @stephencelis  and @mbrandonw was that it would potentially be very expensive because you would have to basically run the wrapped parser against the input, character by character, and that parser could be very expensive.

I think it could be quite useful, but it's definitely a performance risk.
